src/index.js

```diff
@@ -0,0 +1,154 @@
+
+      axios
+        .get('http://127.0.0.1:5000/weather', {
+          params: {
+            lat: latitude,
+            lon: longitude,
+            units: 'imperial', // added extra param in weather-report-proxy-server
+          },
+        })
+        .then((response) => {
+          const weatherResult = response.data.current['temp'];
+          document.getElementById('display-temp').textContent =
+            Math.round(weatherResult);
+          changeTempColor();
+          changeLandscape();
+        });
+    })
+    .catch((error) => {
+      if (error instanceof TypeError) {
+        alert('Please enter a valid city.');
+        return false;
+      } else {
+        console.log('error:', error.response.data);
+        throw error;
+      }
+    });
+};
+
+const changeSky = () => {
+  const skySelection = document.getElementById('sky');
+  const chosenSky =
+    skySelection.options[skySelection.selectedIndex].textContent;
+  let skyDisplay = document.getElementById('sky-emojis');
+  if (chosenSky === 'Sunny') {
+    skyDisplay.textContent = '🌈🌤🌈🌤🌈🌤🌈🌤🌈';
+  } else if (chosenSky === 'Cloudy') {
+    skyDisplay.textContent = '🌤🌥🌤🌥🌤🌥🌤🌥🌤🌥';
+  } else if (chosenSky === 'Rainy') {
+    skyDisplay.textContent = '🌧🌦🌧🌦🌧🌦🌧🌦🌧🌦';
+  } else if (chosenSky === 'Snowy') {
+    skyDisplay.textContent = '🌨🌨🌨🌨🌨🌨🌨🌨🌨🌨';
+  }
+};
+
+const resetCityNameAndTemp = () => {
+  document.getElementById('city-name-display').textContent = 'Seattle';
+  document.getElementById('display-temp').textContent = '77';
+  const temperature = document.getElementById('display-temp');
+  temperature.style.color = 'blue';
+  const landscape = document.getElementById('landscape');
+  landscape.textContent = '🌸🌿🌼🍀🌷🌻🌿🍀🌱🍀🌻🌷';
+};
+
+const clearInputField = () => {
+  document.getElementById('city-name').value = '';
+};
+
+const registerEventHandlers = () => {
+  const increaseTemp = document.getElementById('adjust-temp-up');
+  const decreaseTemp = document.getElementById('adjust-temp-down');
+  const cityName = document.querySelector('input');
+  const cityRealTimeTemp = document.getElementById('real-time-temp');
+  const changingSky = document.getElementById('sky-emojis');
+  const resettingCity = document.getElementById('reset-button');
+  increaseTemp.addEventListener('click', () => {
+    raiseTemp();
+  });
+  decreaseTemp.addEventListener('click', () => {
+    lowerTemp();
+  });
+  cityName.addEventListener('keyup', () => {
+    updateCityName();
+  });
+  cityRealTimeTemp.addEventListener('click', () => {
+    realTimeTemp();
+    clearInputField();
+  });
+  changingSky.addEventListener('change', () => {
+    changeSky();
+  });
+  resettingCity.addEventListener('click', () => {
+    resetCityNameAndTemp();
+    clearInputField();
+  });
+};
+
+// tried to add this, but messed up my display temperature number (would only increase or decrease by odd numbers)
+// document.addEventListener('DOMContentLoaded', registerEventHandlers);
```

Comment:
My guess for reason behind this behavior is because in the HTML, the event listener is added with `onclick="raiseTemp()"` for example, so when the button is clicked with the second event handler registration, the raiseTemp function is called twice?